extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -134,46 +134,37 @@ public Optional<InputStream> streamTaskLog(String taskid, long offset)
   @Override
   public ListenableFuture<TaskStatus> run(Task task)
   {
-    return tasks.computeIfAbsent(
-        task.getId(), k -> new KubernetesWorkItem(task, exec.submit(() -> runTask(task)))
-    ).getResult();
+    return runOrJoinTask(task, true);
   }
 
-  protected ListenableFuture<TaskStatus> joinAsync(Task task)
+  protected ListenableFuture<TaskStatus> runOrJoinTask(Task task, boolean run)
   {
-    return tasks.computeIfAbsent(
-        task.getId(), k -> new KubernetesWorkItem(task, exec.submit(() -> joinTask(task)))
-    ).getResult();
-  }
-
-  private TaskStatus runTask(Task task)
-  {
-    return doTask(task, true);
-  }
-
-  private TaskStatus joinTask(Task task)
-  {
-    return doTask(task, false);
+    synchronized (tasks) {
+      tasks.computeIfAbsent(task.getId(), k -> new KubernetesWorkItem(task, exec.submit(() -> doTask(task, run))));
+      return tasks.get(task.getId()).getResult();
```

Comment:
this can be a single chained statement. Also it would look more readable if the arguments to `computeIfAbsent` were on a different line as in the original method.

Reply:
Updated.